.github/workflows/auto_docs.yml

```diff
@@ -0,0 +1,30 @@
+name: Automatic Documentation for Launcher
+
+on:
+  push:
+    branches:
+      - main
+
+jobs:
+  update_output:
+    runs-on: ubuntu-latest
+    
+    steps:
+    - name: Checkout code
+      uses: actions/checkout@v2
+    
+    - name: Run TGI Launcher docs
+      id: run-launcher-help
+      run: |
+        output=$(text-generation-launcher --help)
+        echo "::set-output name=output::$output"
+    
+    - name: Update Docs
+      run: |
+        output="${{ steps.run-launcher-help.outputs.output }}"
+        
+        echo $output >> ../../docs/source/basic_tutorials/launcher.md
+
+        git add launcher.md
+        git commit -m "Update launcher.md"
+        git push
```

Comment:
This would be doing a new push to main, but the workflow triggers with every new push. Wouldn't this trigger an infinite loop? (not sure)

Reply:
Personally, I think it would be better to make this check that `launcher.md` is up to date (as we do with `black` or `isort`, they just check but don't change files), and fail if running `text-generation-launcher --help >> launcher.md` would change the contents of `launcher.md`, hence forcing users to run this command locally. 
(no strong opinion, I think @Narsil had suggested the approach of pushing post-merge)

Reply:
I setup what you said of having a check instead of force-push, and it also works. Both are fine, but indeed it could trigger infinite loop I don't knwo how GH handles those.

Reply:
Isn't this always be up to date since it runs every time there's a change in `main.rs`? 
```
rm ../../docs/source/basic_tutorials/launcher.md
echo '```' >> ../../docs/source/basic_tutorials/launcher.md
echo $output >> ../../docs/source/basic_tutorials/launcher.md
echo '```' >> ../../docs/source/basic_tutorials/launcher.md
```
so whenever there's a change in the file it will always remove the launcher.md and then create new one and post output. Is this what you mean? 

About when to push, I agree I should change it to post-merge. I've never done it in GHA so I'm looking into it.

Reply:
I will think more on this, I can't push this way either :D

Reply:
My suggestion was to, rather than GA updating `launcher.md` directly, just have GA check if the changes in `main.rs` would trigger a change in `launcher.md`. E.g. you can do this by checking if output == content in launcher.md. If it's the same, nothing happens. If it's different, then the GA workflow fails. 

This means that users, not GA, will be forced to run the command to update the doc (it could be a bash script encapsulating everything as well). (again, no strong opinion, it has its pros and cons)

Reply:
I did it like so: (also started working in another PR since in this fork the action is not triggered)
 ```
   - name: Update Docs
      run: |
        output="${{ steps.run-launcher-help.outputs.output }}"
        launcher_content=$(/docs/source/basic_tutorials/launcher.md)

        if [ "$launcher_content" != "$output" ]; then
          rm /docs/source/basic_tutorials/launcher.md
          echo '```' >> /docs/source/basic_tutorials/launcher.md
          echo $output >> /docs/source/basic_tutorials/launcher.md
          echo '```' >> /docs/source/basic_tutorials/launcher.md
          echo "Docs are updated!"

          git add launcher.md
          git commit -m "Update launcher.md"
          git push
```